src/entries/popup/hooks/useTransactionListForPendingTxs.ts

```diff
@@ -23,68 +20,65 @@ import { isLowerCaseMatch } from '~/core/utils/strings';
 export const useTransactionListForPendingTxs = () => {
   const { currentAddress: address } = useCurrentAddressStore();
   const { currentCurrency: currency } = useCurrentCurrencyStore();
-  const { getPendingTransactions } = usePendingTransactionsStore();
-  const pendingTransactions = getPendingTransactions({ address });
+  // const { getPendingTransactions } = usePendingTransactionsStore();
   const { testnetMode } = useTestnetModeStore();
 
   const supportedChainIds = getBackendSupportedChains({ testnetMode }).map(
     ({ id }) => id,
   );
 
-  useConsolidatedTransactions(
-    {
-      address,
-      currency,
-      userChainIds: supportedChainIds,
-      testnetMode,
-    },
-    {
-      onSuccess: (data) => {
-        if (data?.pages) {
-          const latestTransactions = data.pages
-            .map((p) => p.transactions)
-            .flat()
-            .filter((t) => isLowerCaseMatch(t.from, address))
-            .reduce(
-              (latestTxMap, currentTx) => {
-                const currentChain = currentTx?.chainId;
-                if (currentChain) {
-                  const latestTx = latestTxMap.get(currentChain);
-                  if (!latestTx) {
-                    latestTxMap.set(currentChain, currentTx);
-                  }
-                }
-                return latestTxMap;
-              },
-              new Map(
-                getSupportedChainIds().map((chain) => [
-                  chain,
-                  null as RainbowTransaction | null,
-                ]),
-              ),
-            );
-          watchForPendingTransactionsReportedByRainbowBackend({
-            currentAddress: address,
-            pendingTransactions,
-            latestTransactions,
-          });
-        }
-      },
-    },
-  );
+  const { data } = useConsolidatedTransactions({
+    address,
+    currency,
+    userChainIds: supportedChainIds,
+    testnetMode,
+  });
+
+  useEffect(() => {
+    if (!data?.pages) return;
+    const latestTransactions = data.pages
+      .map((p) => p.transactions)
+      .flat()
+      .filter((t) => isLowerCaseMatch(t.from, address))
+      .reduce(
+        (latestTxMap, currentTx) => {
+          const currentChain = currentTx?.chainId;
+          if (currentChain) {
+            const latestTx = latestTxMap.get(currentChain);
+            if (!latestTx) {
+              latestTxMap.set(currentChain, currentTx);
+            }
+          }
+          return latestTxMap;
+        },
+        new Map(
+          getSupportedChainIds().map((chain) => [
+            chain,
+            null as RainbowTransaction | null,
+          ]),
+        ),
+      );
+    watchForPendingTransactionsReportedByRainbowBackend({
+      currentAddress: address,
+      // pendingTransactions: getPendingTransactions({ address }),
+      latestTransactions,
+    });
+  }, [address, data?.pages]);
```

Comment:
onSuccess is deprecated on v5, maybe this could be one issue, the onSuccess don't run again when returning from the cache maybe some racing condition idk
https://tkdodo.eu/blog/breaking-react-querys-api-on-purpose